client/ayon_houdini/plugins/publish/collect_frames_fix.py

```diff
@@ -0,0 +1,101 @@
+import pyblish.api
+import ayon_api
+
+from ayon_core.lib.attribute_definitions import (
+    TextDef,
+    BoolDef
+)
+from ayon_core.pipeline.publish import AYONPyblishPluginMixin
+
+from ayon_houdini.api import plugin
+
+
+class CollectFramesFixDef(
+    plugin.HoudiniInstancePlugin,
+    AYONPyblishPluginMixin
+):
+    """Provides text field to insert frame(s) to be rerendered.
+
+    Published files of last version of an instance product are collected into
+    instance.data["last_version_published_files"]. All these but frames
+    mentioned in text field will be reused for new version.
+    """
+    order = pyblish.api.CollectorOrder + 0.495
+    label = "Collect Frames to Fix"
+    targets = ["local"]
+    families = ["*"]
+
+    rewrite_version_enable = False
+
+    def process(self, instance):
+        attribute_values = self.get_attr_values_from_data(instance.data)
+        frames_to_fix = attribute_values.get("frames_to_fix")
+
+        rewrite_version = attribute_values.get("rewrite_version")
+
+        if not frames_to_fix:
+            return
+
+        instance.data["frames_to_fix"] = frames_to_fix
+
+        product_name = instance.data["productName"]
+        folder_entity = instance.data["folderEntity"]
+
+        project_entity = instance.data["projectEntity"]
+        project_name = project_entity["name"]
+
+        version_entity = ayon_api.get_last_version_by_product_name(
+            project_name,
+            product_name,
+            folder_entity["id"]
+        )
+        if not version_entity:
+            self.log.warning(
+                "No last version found, re-render not possible"
+            )
+            return
+
+        representations = ayon_api.get_representations(
+            project_name, version_ids={version_entity["id"]}
+        )
+        published_files = []
+        for repre in representations:
+            # TODO get product type from product entity instead of
+            #   representation 'context' data.
+            repre_context = repre["context"]
+            product_type = repre_context.get("product", {}).get("type")
+            if not product_type:
+                product_type = repre_context.get("family")
+            if "*" not in self.families and product_type not in self.families:
+                continue
+
+            for file_info in repre.get("files"):
+                published_files.append(file_info["path"])
```

Comment:
This seems fishy - isn't the product type defined on the product or the version? Getting it from the `context` of the representation seems a bit odd since that context is just the context for the template data?

Reply:
I wish if I was able to tell. I desperately copied it.
For reference,  https://github.com/ynput/ayon-core/pull/839#issuecomment-2293777570

Reply:
@BigRoy This collector is inherited from the code which lives in the ayon-core where it is used for collecting frames fix in Nuke, which also has the same functionality of overwriting the last publish. You can find this PR helpful: https://github.com/ynput/ayon-core/pull/839